.github/workflows/release.yaml

```diff
@@ -0,0 +1,37 @@
+name: generate-release
+# This workflow will generate changelog and release notes.
+# Source: https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/.github/workflows/release.yml
+
+on:
+  workflow_dispatch:
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.x
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
```

Comment:
**NOTE:** this uses a [deploy key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/managing-deploy-keys#deploy-keys) to commit regenerated code to the `main` branch before release.

The `metal-go` version is hard-coded in the User Agent and the docs, so when we publish a new release, we have to first regenerate the code with the correct version number, and then tag that new commit for the release.

The default GitHub Actions token cannot commit to a protected branch.  We could use a Personal Access Token, but IMO it's not fun to have an individual's access token tied to an automated workflow.

As used here, the deploy key is extremely limited in scope (a deploy key is only usable for git operations, not for GitHub API operations, it has write access to any repo that uses the same deploy key, and the deploy key only exists on this repo). 

In order to run this workflow, someone with admin permissions has to first go in and turn off the `Do not allow bypassing the above settings` option in the branch protection settings (after release it can be turned on manually or wait for the automation to do it).  Ideally, we could just turn that setting off (and potentially create more granular GitHub teams to better control access levels).

Reply:
My actions may be chased by robot rules, but this has been disabled for science.

Reply:
I...don't know what that means?